File: benchmarks/vire-lang/mandelbrot.cpp
#include <cstdio>
#include <cstdint>
int main() {
    int64_t n = 2000, count = 0;
    for (int64_t py = 0; py < n; py++) for (int64_t px = 0; px < n; px++) {
        double cr = px * 3.0 / n - 2.0, ci = py * 2.0 / n - 1.0, zr = 0, zi = 0;
        int64_t esc = 0, i = 0;
        while (i < 50) { double zr2 = zr*zr - zi*zi + cr; zi = 2.0*zr*zi + ci; zr = zr2;
            if (zr*zr + zi*zi > 4.0) { esc = 1; i = 50; } else i++; }
        count += 1 - esc;
    }
    printf("%ld\n", count);
}

File: benchmarks/vire-lang/mandelbrot.rs
fn main() {
    let n = 2000i64;
    let mut count = 0i64;
    for py in 0..n {
        for px in 0..n {
            let cr = px as f64 * 3.0 / n as f64 - 2.0;
            let ci = py as f64 * 2.0 / n as f64 - 1.0;
            let (mut zr, mut zi) = (0.0f64, 0.0f64);
            let mut esc = 0i64;
            let mut i = 0;
            while i < 50 {
                let zr2 = zr*zr - zi*zi + cr;
                zi = 2.0*zr*zi + ci;
                zr = zr2;
                if zr*zr + zi*zi > 4.0 { esc = 1; i = 50; } else { i += 1; }
            }
            count += 1 - esc;
        }
    }
    println!("{}", count);
}
